### ISSUE TITLE
MSCOCO - Class-wise image download or Class-wise Image separation.

### ISSUE THREAD

**Issue by chandanv2** [cocodataset/cocoapi#120]
How to download or seperate MSCOCO images of a particular class? Are there any scripts to do that?

**Comment by salihkaragoz** [comment]
First choose which class do you need 
[categories](https://gist.github.com/salihkaragoz/8c3291b55abee745389de150df94bbca)

Then you can use cocoapi to get specific categories id image' instance.
For example backpack ID : 27

```
from pycocotools.coco import COCO

coco_train = COCO('instances_train2017.json')
coco_img_ids = COCO.getCatIds(catIds=27)
```

**Comment by chandanv2** [comment]
Thanks!

**Comment by syedmutti** [comment]
Hey, Does the last line download or seperate images ? 
Do I need to have MSCOCO Dataset already downloaded ?

**Comment by salihkaragoz** [comment]
@syedmutti Class-wise downloads not possible. you should first download images. if you want to only get ids of some class. you can do with above commands. it gives you the identities of images. you just need the annotations of coco dataset such as 'instances_train_2017.json'.

**Comment by syedmutti** [comment]
@salihkaragoz  Thanks for clearifying, I have downloaded and seperated the image ids.
But, I am wondering if it is possible to really seperate the images and extract annotations as I wish to add few more classes in my DataSet that are not included in MSCOCO and train a network on it.
Also, It would give me freedom to change the annotations to a different format that might be compatible with darknet or so.

Any suggestions on what would be the best way to achieve that ?

**Comment by AlphaArslan** [comment]
guys!! 
I'm trying to separate images of specific classes and move them to new folder 
how can I do that ?
..
I'm trying to use COCO to train a model to detect some classes (all of them inside COCO)
thanks!

**Comment by Yoan-D** [comment]
### Assuming that the dataset has already been downloaded. Here is an example of getting all images from the training dataset for the category _horse_.  You can easily generalize this approach to separate all categories into new folders

    from pycocotools import coco
    import os
    import shutil

    coco = coco.COCO('annotations/instances_train2017.json')  # note this only refers to  the 
    training set and not the validation set

    categories = coco.loadCats(coco.getCatIds())

    names = [cat['name'] for cat in categories]

    print("Available categories: ")
    for index, n in enumerate(names):
        print(index, n)

    category_ids = coco.getCatIds(catNms=['horse'])
    image_ids = coco.getImgIds(catIds=category_ids)
    images = coco.loadImgs(image_ids)

    horse_file_names = []
    for img in images:
        horse_file_names.append(img['file_name'])

    # Create folder for the images of the selected category   
    
    os.mkdir('horse_images')

    for filename in os.listdir('train2017'):
        if filename in horse_file_names:
            shutil.copy(os.path.join('train2017', filename), 'horse_images')

    print('Done creating data subset with horse images....')

**Comment by adarsh0906** [comment]
how to recreate the instance.json format for some category say car,bus,truck,motorcycle

**Comment by mesakarghm** [comment]
> how to recreate the instance.json format for some category say car,bus,truck,motorcycle

Anyone got solution to this? I need to seperate the pictures as well the instance.json. As I understand, the above provided code only separates the images from the dataset.

**Comment by sag3011kumbari** [comment]
> > how to recreate the instance.json format for some category say car,bus,truck,motorcycle
> 
> Anyone got solution to this? I need to seperate the pictures as well the instance.json. As I understand, the above provided code only separates the images from the dataset.

I too need it please if anyone knows about it please tell
